main.tf

```diff
@@ -572,8 +592,9 @@ resource "aws_iam_role_policy" "cloud_watch_alarm_filter" {
 }
 
 resource "aws_lambda_function" "cloud_watch_alarm_filter" {
+  depends_on = [data.archive_file.lambda]
```

Comment:
can you explain why this is needed?

Reply:
I was having a problem with archive files being created, turns out they now get created in plan not apply and wasn't making in between CI pipeline stages, also why I have now set to be created in the root of the terraform that calls the module.